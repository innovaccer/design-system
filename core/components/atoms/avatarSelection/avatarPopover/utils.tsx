import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>,
  listRef?: any,
  withSearch?: boolean,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'Enter':
      handleEnterKey(focusedOption);
      setHighlightLastItem?.(false);
      setHighlightFirstItem?.(false);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }
};

const handleEnterKey = (focusedOption: Element | undefined) => {
  (focusedOption as HTMLElement)?.click();
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  const listItems = listRef?.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else if (
    (withSearch && index === 0 && direction === 'up') ||
    (withSearch && index === listItems.length - 1 && direction === 'down')
  ) {
    const searchInput = listRef.current.querySelector('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput.focus();
    setFocusedOption && setFocusedOption(searchInput);
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;

    const targetOption = listItems[index];

    (targetOption as HTMLElement).focus();
    setFocusedOption && setFocusedOption(targetOption);
    targetOption.scrollIntoView({ block: 'center' });
  }
};

export const handleInputKeyDown = (
  event: React.KeyboardEvent,
  listRef: any,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      targetOption = listItems[listItems.length - 1];
      break;
    case 'ArrowDown':
      event.preventDefault();
      targetOption = listItems[0];
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }

  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
