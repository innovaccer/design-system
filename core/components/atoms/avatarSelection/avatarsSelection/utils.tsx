import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>
) => {
  switch (event.key) {
    case 'Enter':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowDown':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowUp':
      setOpenPopover?.(true);
      setHighlightLastItem?.(true);
      break;
    default:
      break;
  }
};

export const focusListItem = (
  position: string,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  if (withSearch && position === 'down') {
    const searchInput = listRef.current?.querySelectorAll('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput && searchInput[0]?.focus();
    setFocusedOption?.(searchInput);
    return;
  }

  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  if (position === 'down') {
    targetOption = listItems?.[0];
  } else {
    targetOption = listItems?.[listItems.length - 1];
  }
  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
