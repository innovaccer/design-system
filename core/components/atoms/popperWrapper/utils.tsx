// Helper function to convert CSS class names to style objects
export const createStyleFromClass = (className: string) => {
  if (!className) return {};

  // Create a temporary element to get computed styles
  const tempElement = document.createElement('div');
  tempElement.className = className;
  document.body.appendChild(tempElement);

  const computedStyle = window.getComputedStyle(tempElement);
  const style: React.CSSProperties = {};

  // Extract relevant animation properties
  const animation = computedStyle.animation;
  const animationDuration = computedStyle.animationDuration;
  const animationTimingFunction = computedStyle.animationTimingFunction;
  const animationFillMode = computedStyle.animationFillMode;
  const opacity = computedStyle.opacity;
  const transform = computedStyle.transform;

  if (animation && animation !== 'none') {
    style.animation = animation;
  }
  if (animationDuration && animationDuration !== '0s') {
    style.animationDuration = animationDuration;
  }
  if (animationTimingFunction && animationTimingFunction !== 'ease') {
    style.animationTimingFunction = animationTimingFunction;
  }
  if (animationFillMode && animationFillMode !== 'none') {
    style.animationFillMode = animationFillMode;
  }
  if (opacity && opacity !== '1') {
    style.opacity = opacity;
  }
  if (transform && transform !== 'none') {
    style.transform = transform;
  }

  document.body.removeChild(tempElement);
  return style;
};
