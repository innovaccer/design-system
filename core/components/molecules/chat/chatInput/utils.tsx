import { MessageType } from './ChatInput';

export const positionMentionPopup = (
  textareaRef: React.RefObject<HTMLDivElement>,
  setMentionPosition: React.Dispatch<React.SetStateAction<{ top: number; left: number }>>
) => {
  if (!textareaRef.current) return;

  const selection = window.getSelection();
  if (!selection || !selection.rangeCount) return;

  const range = selection.getRangeAt(0);
  const rect = range.getBoundingClientRect();
  const editorRect = textareaRef.current.getBoundingClientRect();

  // Calculate position based on the cursor position
  let top = rect.bottom + window.scrollY;
  let left = rect.left + window.scrollX;

  // Adjust position if it would go off-screen
  if (top + 150 > window.innerHeight) {
    top = rect.top + window.scrollY - 150;
  }
  if (left + 200 > window.innerWidth) {
    left = window.innerWidth - 200;
  }

  // Adjust position relative to the editor
  top -= editorRect.top + window.scrollY - 12;
  left -= editorRect.left + window.scrollX - 8;

  setMentionPosition({ top, left });
};

export const extractMessageData = (textareaRef: React.RefObject<HTMLDivElement>): MessageType[] => {
  const elements = textareaRef.current?.childNodes;

  return Array.from(elements || []).map((item: any) => {
    if (item?.dataset?.type === 'mention') {
      const mentionValue = JSON.parse(item.dataset.content);
      return { type: 'mention', content: mentionValue?.data, nodeName: item.nodeName };
    }

    // handles new line by pressing enter
    else if (item.innerText === '\n' && item.nodeName === 'DIV') {
      return { type: 'newline', content: '\n', nodeName: item.nodeName };
    }

    // handles line break by pressing shift + enter
    else if (item.nodeName === 'BR') {
      return { type: 'linebreak', content: '<br>', nodeName: item.nodeName };
    }

    return { type: 'text', content: item.innerText || item.textContent, nodeName: item.nodeName };
  });
};
