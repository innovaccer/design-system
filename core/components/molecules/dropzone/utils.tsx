const isIe = (userAgent: string) => {
  return userAgent.indexOf('MSIE') !== -1 || userAgent.indexOf('Trident/') !== -1;
};

const isEdge = (userAgent: string) => {
  return userAgent.indexOf('Edge/') !== -1;
};

export const accepts = (file: File, acceptedFiles?: string | string[]) => {
  if (file && acceptedFiles) {
    const acceptedFilesArray = Array.isArray(acceptedFiles) ? acceptedFiles : acceptedFiles.split(',');
    const fileName = file.name || '';
    const mimeType = (file.type || '').toLowerCase();
    const baseMimeType = mimeType.replace(/\/.*$/, '');

    if (!mimeType) {
      return true;
    }

    return acceptedFilesArray.some((type) => {
      const validType = type.trim().toLowerCase();
      if (validType.charAt(0) === '.') {
        return fileName.toLowerCase().endsWith(validType);
      }
      if (validType.endsWith('/*')) {
        return baseMimeType === validType.replace(/\/.*$/, '');
      }
      return mimeType === validType;
    });
  }
  return true;
};

export const isPropagationStopped = (event: any) => {
  if (typeof event.isPropagationStopped === 'function') {
    return event.isPropagationStopped();
  }
  if (typeof event.cancelBubble !== 'undefined') {
    return event.cancelBubble;
  }
  return false;
};

export const isEvtWithFiles = (event: any) => {
  if (!event.dataTransfer) {
    return !!event.target && !!event.target.files;
  }

  return Array.prototype.some.call(
    event.dataTransfer.types,
    (type) => type === 'Files' || type === 'application/x-moz-file'
  );
};

export const isKindFile = (item: null | Record<string, unknown>) => {
  return typeof item === 'object' && item !== null && item.kind === 'file';
};

export const onDocumentDragOver = (event: DragEvent) => {
  event.preventDefault();
};

export const isIeOrEdge = (userAgent = window.navigator.userAgent) => {
  return isIe(userAgent) || isEdge(userAgent);
};

type functype = (event: Event, ...args: unknown[]) => void;

export const composeEventHandlers = (...fns: functype[]) => {
  return (event: Event, ...args: unknown[]) =>
    fns.some((fn) => {
      if (!isPropagationStopped(event) && fn) {
        fn(event, ...args);
      }
      return isPropagationStopped(event);
    });
};

export const reducer = (state: any, action: any) => {
  switch (action.type) {
    case 'focus':
      return {
        ...state,
        isFocused: true,
      };
    case 'blur':
      return {
        ...state,
        isFocused: false,
      };
    case 'openDialog':
      return {
        ...state,
        isFileDialogActive: true,
      };
    case 'closeDialog':
      return {
        ...state,
        isFileDialogActive: false,
      };
    case 'setDraggedFiles':
      /* eslint no-case-declarations: 0 */
      const { isDragActive, draggedFiles } = action;
      return {
        ...state,
        draggedFiles,
        isDragActive,
      };
    case 'setFiles':
      return {
        ...state,
        acceptedFiles: action.acceptedFiles,
        fileRejections: action.fileRejections,
      };
    case 'reset':
      return {
        ...state,
        isFileDialogActive: false,
        isDragActive: false,
        draggedFiles: [],
        acceptedFiles: [],
        fileRejections: [],
      };
    default:
      return state;
  }
};
