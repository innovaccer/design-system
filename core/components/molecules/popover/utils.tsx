import { Placement } from '@floating-ui/react';
import { PositionType as Position } from '@/common.type';

export const mapPositionToPlacement = (position: Position): Placement => {
  const positionMap: Record<Position, Placement> = {
    auto: 'bottom',
    'auto-start': 'bottom-start',
    'auto-end': 'bottom-end',
    'top-start': 'top-start',
    top: 'top',
    'top-end': 'top-end',
    'right-start': 'right-start',
    right: 'right',
    'right-end': 'right-end',
    'bottom-end': 'bottom-end',
    bottom: 'bottom',
    'bottom-start': 'bottom-start',
    'left-end': 'left-end',
    left: 'left',
    'left-start': 'left-start',
  };

  return positionMap[position] || 'bottom';
};
