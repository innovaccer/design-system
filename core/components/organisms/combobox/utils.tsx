import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  inputTriggerRef?: any,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>,
  multiSelect?: boolean,
  listRef?: any
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef);
      break;
    case 'Enter':
      handleEnterKey(focusedOption, multiSelect, inputTriggerRef, listRef, setFocusedOption);
      setHighlightLastItem?.(false);
      setHighlightFirstItem?.(false);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      inputTriggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }
};

const handleEnterKey = (
  focusedOption: Element | undefined,
  multiSelect?: boolean,
  inputTriggerRef?: any,
  listRef?: any,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>
) => {
  (focusedOption as HTMLElement)?.click();
  if (!multiSelect) {
    inputTriggerRef.current.focus();
  } else {
    // to focus first option by default when last option is selected
    const listItems = listRef.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
    const listArr = Array.from(listItems);
    const index = listArr.findIndex((item) => {
      return item == focusedOption;
    });

    if (index === listArr.length - 1) {
      (listItems[0] as HTMLElement).focus();
      setFocusedOption && setFocusedOption(listItems[0]);
      listItems[0].scrollIntoView({ block: 'center' });
    }
  }
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any
) => {
  const listItems = listRef.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;
  }

  const targetOption = listItems[index];

  (targetOption as HTMLElement).focus();
  setFocusedOption && setFocusedOption(targetOption);
  targetOption?.scrollIntoView?.({ block: 'center' });
};
