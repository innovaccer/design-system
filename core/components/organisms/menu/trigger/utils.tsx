import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean | undefined>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      setOpenPopover?.(true);
      setHighlightLastItem?.(true);
      break;
    case 'ArrowDown':
      event.preventDefault();
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'Escape':
    case 'Tab':
      setOpenPopover?.(false);
      break;
    default:
      break;
  }
};

export const focusListItem = (
  position: string,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any
) => {
  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  if (position === 'down') {
    targetOption = listItems?.[0];
  } else {
    targetOption = listItems?.[listItems.length - 1];
  }
  (targetOption as HTMLElement)?.focus();

  if (targetOption && typeof targetOption.scrollIntoView === 'function') {
    (targetOption as HTMLElement)?.scrollIntoView({ block: 'end' });
  }
  setFocusedOption && setFocusedOption(targetOption);
};
