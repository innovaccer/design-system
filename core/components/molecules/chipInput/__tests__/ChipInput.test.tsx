import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ChipInput } from '@/index';
import { ChipInputProps as Props } from '@/index.type';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';

const value = ['1020', '80'];
const placeholder = 'Add value';
const BooleanValue = [true, false];
const FunctionValue = jest.fn();
const chipOptions = {
  clearButton: true,
  onClick: FunctionValue,
};

describe('ChipInput component', () => {
  const mapper = {
    disabled: valueHelper(BooleanValue, { required: true, iterate: true }),
    allowDuplicates: valueHelper(true, { required: true }),
    value: valueHelper(value, { required: true }),
    chipOptions: valueHelper(chipOptions, { required: true }),
    placeholder: valueHelper(placeholder, { required: true }),
    onChange: valueHelper(FunctionValue, { required: true }),
    onBlur: valueHelper(FunctionValue, { required: true }),
    onFocus: valueHelper(FunctionValue, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<ChipInput {...attr} />);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('ChipInput component', () => {
  it('renders input and chips', () => {
    const { getByTestId, getAllByTestId } = render(<ChipInput value={value} />);

    expect(getByTestId('DesignSystem-ChipInput--Input')).toBeInTheDocument();
    expect(getAllByTestId('DesignSystem-ChipInput--Chip')).toHaveLength(value.length);
    expect(getAllByTestId('DesignSystem-ChipInput--Chip')[0].textContent).toMatch(value[0]);
    expect(getAllByTestId('DesignSystem-ChipInput--Chip')[1].textContent).toMatch(value[1]);
  });

  it('initially renders input only', () => {
    const { getByTestId, queryAllByTestId } = render(<ChipInput />);

    expect(getByTestId('DesignSystem-ChipInput--Input')).toBeInTheDocument();
    expect(queryAllByTestId('DesignSystem-ChipInput--Chip')).toHaveLength(0);
  });

  it('renders close icon only if length of chips > 0', () => {
    const { getByTestId, queryByTestId, rerender } = render(<ChipInput value={[]} />);

    expect(queryByTestId('DesignSystem-ChipInput--Icon')).not.toBeInTheDocument();
    rerender(<ChipInput value={value} />);
    expect(getByTestId('DesignSystem-ChipInput--Icon')).toBeInTheDocument();
  });

  it('ChipInput with prop: disabled', () => {
    const { getByTestId, getAllByTestId } = render(<ChipInput value={value} disabled={true} />);

    expect(getByTestId('DesignSystem-ChipInput')).toHaveClass('ChipInput--disabled');
    expect(getAllByTestId('DesignSystem-ChipInput--Chip')[0]).toHaveClass('Chip-input--disabled');
  });

  it('ChipInput with prop: placeholder', () => {
    const { getByTestId } = render(<ChipInput placeholder={placeholder} />);

    expect(getByTestId('DesignSystem-ChipInput--Input')).toHaveAttribute('placeholder', placeholder);
  });

  it('ChipInput with prop allowDuplicate: false', () => {
    const { getByTestId, getAllByTestId } = render(<ChipInput defaultValue={value} onChange={FunctionValue} />);

    const inputComponent = getByTestId('DesignSystem-ChipInput--Input');
    fireEvent.change(inputComponent, { target: { value: value[0] } });
    fireEvent.keyDown(inputComponent, { key: 'Enter' });
    expect(getAllByTestId('DesignSystem-ChipInput--Chip')).toHaveLength(value.length);
  });

  it('ChipInput with prop allowDuplicate: true', () => {
    const { getByTestId, getAllByTestId } = render(
      <ChipInput defaultValue={value} onChange={FunctionValue} allowDuplicates={true} />
    );

    const inputComponent = getByTestId('DesignSystem-ChipInput--Input');
    fireEvent.change(inputComponent, { target: { value: value[0] } });
    fireEvent.keyDown(inputComponent, { key: 'Enter' });
    expect(getAllByTestId('DesignSystem-ChipInput--Chip')).toHaveLength(value.length + 1);
  });
});

describe('Controlled ChipInput component', () => {
  it('Add chip if enter is pressed', () => {
    const newValue = ['1020', '80', '90'];
    const { rerender, getByTestId, getAllByTestId } = render(<ChipInput value={value} onChange={FunctionValue} />);
    expect(getAllByTestId('DesignSystem-ChipInput--Chip')).toHaveLength(value.length);

    const inputComponent = getByTestId('DesignSystem-ChipInput--Input');
    fireEvent.change(inputComponent, { target: { value: '90' } });
    fireEvent.keyDown(inputComponent, { key: 'Enter' });

    expect(FunctionValue).toHaveBeenCalled();
    rerender(<ChipInput value={newValue} onChange={FunctionValue} />);
    expect(getAllByTestId('DesignSystem-ChipInput--Chip')).toHaveLength(newValue.length);
    expect(getAllByTestId('DesignSystem-ChipInput--Chip')[2].textContent).toMatch(newValue[2]);
  });
});

describe('Uncontrolled ChipInput component', () => {
  it('Add chip if enter is pressed', () => {
    const newValue = ['1020', '80', '90'];
    const { getByTestId, getAllByTestId } = render(<ChipInput defaultValue={value} onChange={FunctionValue} />);
    expect(getAllByTestId('DesignSystem-ChipInput--Chip')).toHaveLength(value.length);

    const inputComponent = getByTestId('DesignSystem-ChipInput--Input');
    fireEvent.change(inputComponent, { target: { value: '90' } });
    fireEvent.keyDown(inputComponent, { key: 'Enter' });

    expect(FunctionValue).toHaveBeenCalled();
    expect(getAllByTestId('DesignSystem-ChipInput--Chip')).toHaveLength(newValue.length);
    expect(getAllByTestId('DesignSystem-ChipInput--Chip')[2].textContent).toMatch(newValue[2]);
  });

  it('Delete chip if backspace is pressed', () => {
    const newValue = ['1020'];
    const { getByTestId, getAllByTestId } = render(<ChipInput defaultValue={value} onChange={FunctionValue} />);
    expect(getAllByTestId('DesignSystem-ChipInput--Chip')).toHaveLength(value.length);

    const inputComponent = getByTestId('DesignSystem-ChipInput--Input');
    fireEvent.keyDown(inputComponent, { key: 'Backspace' });

    expect(FunctionValue).toHaveBeenCalled();
    expect(getAllByTestId('DesignSystem-ChipInput--Chip')).toHaveLength(newValue.length);
    expect(getAllByTestId('DesignSystem-ChipInput--Chip')[0].textContent).toMatch(newValue[0]);
  });

  it('Delete all chips if close icon is clicked', () => {
    const newValue: string[] = [];
    const { getByTestId, getAllByTestId, queryAllByTestId } = render(
      <ChipInput defaultValue={value} onChange={FunctionValue} />
    );
    expect(getAllByTestId('DesignSystem-ChipInput--Chip')).toHaveLength(value.length);

    const closeIcon = getByTestId('DesignSystem-ChipInput--Icon');
    fireEvent.click(closeIcon);

    expect(FunctionValue).toHaveBeenCalled();
    expect(queryAllByTestId('DesignSystem-ChipInput--Chip')).toHaveLength(newValue.length);
  });
});
