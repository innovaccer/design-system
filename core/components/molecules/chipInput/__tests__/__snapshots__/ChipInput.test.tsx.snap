// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChipInput component 
  disabled: false, allowDuplicates: true, value: ["1020","80"], chipOptions: {"clearButton":true,"onClick":"[Function]"}, placeholder: "Add value", onChange: "[Function]", onBlur: "[Function]", onFocus: "[Function]"
 1`] = `
<body>
  <div>
    <div
      class="ChipInput ChipInput--withChips"
      data-test="DesignSystem-ChipInput"
    >
      <div
        class="ChipInput-wrapper"
      >
        <div
          class="Chip-wrapper Chip Chip--input my-2 mx-2"
          data-test="DesignSystem-ChipInput--Chip"
        >
          <span
            class="Text Text--default Text--regular"
            data-test="DesignSystem-GenericChip--Text"
          >
            1020
          </span>
          <i
            class="material-icons material-icons-round Icon Chip-icon Chip-icon--right cursor-pointer"
            data-test="DesignSystem-GenericChip--clearButton"
            role="button"
            style="font-size: 16px; width: 16px;"
            tabindex="0"
          >
            clear_round
          </i>
        </div>
        <div
          class="Chip-wrapper Chip Chip--input my-2 mx-2"
          data-test="DesignSystem-ChipInput--Chip"
        >
          <span
            class="Text Text--default Text--regular"
            data-test="DesignSystem-GenericChip--Text"
          >
            80
          </span>
          <i
            class="material-icons material-icons-round Icon Chip-icon Chip-icon--right cursor-pointer"
            data-test="DesignSystem-GenericChip--clearButton"
            role="button"
            style="font-size: 16px; width: 16px;"
            tabindex="0"
          >
            clear_round
          </i>
        </div>
        <input
          class="ChipInput-input"
          data-test="DesignSystem-ChipInput--Input"
          placeholder="Add value"
          value=""
        />
      </div>
      <i
        class="material-icons material-icons-round Icon Icon--subtle ChipInput-icon"
        data-test="DesignSystem-ChipInput--Icon"
        role="button"
        style="font-size: 16px; width: 16px;"
        tabindex="0"
      >
        close_round
      </i>
    </div>
  </div>
</body>
`;

exports[`ChipInput component 
  disabled: true, allowDuplicates: true, value: ["1020","80"], chipOptions: {"clearButton":true,"onClick":"[Function]"}, placeholder: "Add value", onChange: "[Function]", onBlur: "[Function]", onFocus: "[Function]"
 1`] = `
<body>
  <div>
    <div
      class="ChipInput ChipInput--disabled ChipInput--withChips"
      data-test="DesignSystem-ChipInput"
    >
      <div
        class="ChipInput-wrapper"
      >
        <div
          class="Chip-wrapper Chip Chip-input--disabled my-2 mx-2"
          data-test="DesignSystem-ChipInput--Chip"
        >
          <span
            class="Text Text--disabled Text--regular"
            data-test="DesignSystem-GenericChip--Text"
          >
            1020
          </span>
          <i
            class="material-icons material-icons-round Icon Icon--disabled Chip-icon Chip-icon--right"
            data-test="DesignSystem-GenericChip--clearButton"
            role="button"
            style="font-size: 16px; width: 16px;"
            tabindex="0"
          >
            clear_round
          </i>
        </div>
        <div
          class="Chip-wrapper Chip Chip-input--disabled my-2 mx-2"
          data-test="DesignSystem-ChipInput--Chip"
        >
          <span
            class="Text Text--disabled Text--regular"
            data-test="DesignSystem-GenericChip--Text"
          >
            80
          </span>
          <i
            class="material-icons material-icons-round Icon Icon--disabled Chip-icon Chip-icon--right"
            data-test="DesignSystem-GenericChip--clearButton"
            role="button"
            style="font-size: 16px; width: 16px;"
            tabindex="0"
          >
            clear_round
          </i>
        </div>
        <input
          class="ChipInput-input"
          data-test="DesignSystem-ChipInput--Input"
          disabled=""
          placeholder="Add value"
          value=""
        />
      </div>
      <i
        class="material-icons material-icons-round Icon Icon--subtle ChipInput-icon"
        data-test="DesignSystem-ChipInput--Icon"
        role="button"
        style="font-size: 16px; width: 16px;"
        tabindex="0"
      >
        close_round
      </i>
    </div>
  </div>
</body>
`;
